### PR TITLE
fix(core): data loss on enable cloud

### PR DESCRIPTION
**Pull request by EYHN**

### CODE REVIEW

**Comment by EYHN** [comment]
* **#8452** <a href="https://app.graphite.dev/github/pr/toeverything/AFFiNE/8452?utm_source=stack-comment-icon" target="_blank">Graphite</a> 👈
* `canary`

This stack of pull requests is managed by Graphite. <a href="https://stacking.dev/?utm_source=stack-comment">Learn more about stacking.</a>
<h2></h2>

 Join @EYHN and the rest of your teammates on <a href="https://graphite.dev?utm-source=stack-comment">Graphite <b>Graphite</b></a>

File: packages/common/infra/src/modules/workspace/services/transform.ts [hunk 1]
```diff
@@ -28,29 +28,31 @@ export class WorkspaceTransformService extends Service {
...
+        if (rootDocBinary) {
+          applyUpdate(docCollection.doc, rootDocBinary);
+        }
+
+        for (const subdoc of docCollection.doc.getSubdocs()) {
+          const subdocBinary = await localDocStorage.doc.get(subdoc.guid);
+          if (subdocBinary) {
+            applyUpdate(subdoc, subdocBinary);
           }
         }
```
Comment (pengx17) [inline packages/common/infra/src/modules/workspace/services/transform.ts:49]: parallel this to speed it up?

**Review by pengx17 — APPROVED** [review]

File: packages/common/infra/src/modules/workspace/services/transform.ts (hunk 1 above)
Comment (forehalo) [reply packages/common/infra/src/modules/workspace/services/transform.ts:49]: ui will hang for a very long period

Comment (pengx17) [reply packages/common/infra/src/modules/workspace/services/transform.ts:49]: yeah probably a bad idea

**Pull request merged**